spec/requests/users_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
+  describe 'GET /index' do
+    it 'returns http success' do
+      get '/users/'
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template('users/index')
+      assert_select 'h1', 'Users'
+    end
+
+    it 'returns http success' do
+      get '/users/1'
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template('users/show')
+      assert_select 'h2', 'User data'
+    end
+  end
```

Comment:
[REQUIRED] Good job testing the index and show requests but I think they should be in separate describe blocks like this:

```ruby
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  ...
end
```

[REQUIRED] Kindly ensure that the spec file includes tests that ensure the response body includes the correct placeholder text. Example : `expect(response.body).to include("Welcome to user index page")`